plugins/continuous-toolbox/src/ContinuousToolbox.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Toolbox that uses a continuous scrolling flyout.
+ */
+
+import * as Blockly from 'blockly/core';
+import {ContinuousFlyout} from './ContinuousFlyout';
+
+/**
+ * Class for continuous toolbox.
+ */
+export class ContinuousToolbox extends Blockly.Toolbox {
+  /**
+   * Timeout ID used to prevent refreshing the flyout during extensive block
+   * changes.
+   */
+  private refreshDebouncer?: ReturnType<typeof setTimeout>;
+
+  /**
+   * Initializes the continuous toolbox.
+   */
+  override init() {
+    super.init();
+
+    // Populate the flyout with all blocks and show it immediately.
+    const flyout = this.getFlyout();
+    flyout.show(this.getInitialFlyoutContents());
+
+    this.getWorkspace().addChangeListener((e: Blockly.Events.Abstract) => {
+      if (
+        e.type === Blockly.Events.BLOCK_CREATE ||
+        e.type === Blockly.Events.BLOCK_DELETE ||
+        e.type === Blockly.Events.BLOCK_CHANGE
+      ) {
+        this.refreshSelection();
+      }
+    });
+  }
+
+  /**
+   * Returns the continuous toolbox's flyout.
+   *
+   * @returns The toolbox's flyout.
+   */
+  override getFlyout(): ContinuousFlyout {
+    return super.getFlyout() as ContinuousFlyout;
+  }
+
+  /**
+   * Gets the contents that should be shown in the flyout immediately.
+   * This includes all blocks and labels for each category of block.
+   *
+   * @returns Flyout contents.
+   */
+  private getInitialFlyoutContents(): Blockly.utils.toolbox.FlyoutItemInfoArray {
+    return this.getToolboxItems().flatMap(this.convertToolboxItemToFlyoutItems);
+  }
+
+  /**
+   * Converts a given toolbox item to an array of flyout items, generally a
+   * label followed by the category's blocks.
+   *
+   * @param toolboxItem The toolbox item/category to convert.
+   * @returns An array of flyout items contained in the given toolbox item.
+   */
+  protected convertToolboxItemToFlyoutItems(
+    toolboxItem: Blockly.IToolboxItem,
+  ): Blockly.utils.toolbox.FlyoutItemInfoArray {
+    let contents: Blockly.utils.toolbox.FlyoutItemInfoArray = [];
+    if (toolboxItem instanceof Blockly.ToolboxCategory) {
+      // Create a label node to go at the top of the category
+      contents.push({kind: 'LABEL', text: toolboxItem.getName()});
+      let itemContents = toolboxItem.getContents();
+
+      // Handle custom categories (e.g. variables and functions)
+      if (typeof itemContents === 'string') {
+        itemContents = [
+          {
+            custom: itemContents,
+            kind: 'CATEGORY',
+          },
+        ];
```

Comment:
Unless Prettier objects:
```suggestion
        itemContents = [{custom: itemContents, kind: 'CATEGORY'}];
```

Reply:
Done.